diff --git a/modules/custom/custom/coffee_shop/coffee_shop.services.yml b/modules/custom/custom/coffee_shop/coffee_shop.services.yml
index ed05c68e..47e9d7e3 100644
--- a/modules/custom/custom/coffee_shop/coffee_shop.services.yml
+++ b/modules/custom/custom/coffee_shop/coffee_shop.services.yml
@@ -5,4 +5,7 @@ services:
 #  _defaults:
 #    autowire: true
 
-# just crickets here!
\ No newline at end of file
+  Drupal\coffee_shop\Service\CoffeeMachine:
+    autowire: true
+    arguments:
+      $numberOfScoops: 3
diff --git a/modules/custom/custom/coffee_shop/src/Service/CoffeeMachine.php b/modules/custom/custom/coffee_shop/src/Service/CoffeeMachine.php
index da26f7ce..c7ed42fc 100644
--- a/modules/custom/custom/coffee_shop/src/Service/CoffeeMachine.php
+++ b/modules/custom/custom/coffee_shop/src/Service/CoffeeMachine.php
@@ -3,8 +3,14 @@
 namespace Drupal\coffee_shop\Service;
 
 class CoffeeMachine {
+  private $numberOfScoops;
+
+  public function __construct(int $numberOfScoops) {
+    $this->numberOfScoops = $numberOfScoops;
+  }
+
   public function brew()
   {
-    return 'coffee brewed!';
+    return sprintf('coffee brewed using %d scoops!', $this->numberOfScoops);
   }
 }
