diff --git a/modules/custom/custom/coffee_shop/coffee_shop.services.yml b/modules/custom/custom/coffee_shop/coffee_shop.services.yml
index 880c0873..ecc228e7 100644
--- a/modules/custom/custom/coffee_shop/coffee_shop.services.yml
+++ b/modules/custom/custom/coffee_shop/coffee_shop.services.yml
@@ -1,3 +1,6 @@
 services:
   Drupal\coffee_shop\Service\Barista:
-    arguments: ['@config.factory']
+    arguments: ['@config.factory', '@Drupal\coffee_shop\Service\CoffeeMachine']
+
+  Drupal\coffee_shop\Service\CoffeeMachine:
+    arguments: []
diff --git a/modules/custom/custom/coffee_shop/src/Service/Barista.php b/modules/custom/custom/coffee_shop/src/Service/Barista.php
index 641c7469..1c0b8def 100644
--- a/modules/custom/custom/coffee_shop/src/Service/Barista.php
+++ b/modules/custom/custom/coffee_shop/src/Service/Barista.php
@@ -6,9 +6,11 @@
 
 class Barista {
   private $configFactory;
+  private $coffeeMachine;
 
-  public function __construct(ConfigFactoryInterface $configFactory) {
+  public function __construct(ConfigFactoryInterface $configFactory, CoffeeMachine $coffeeMachine) {
     $this->configFactory = $configFactory;
+    $this->coffeeMachine = $coffeeMachine;
   }
 
   public function prepareDrink($type) {
@@ -20,10 +22,13 @@ public function prepareDrink($type) {
 
     $template = 'A %size% %type% prepared %status%. Enjoy!';
 
+    // some pretend code
+    $message = $this->coffeeMachine->brew();
+
     return strtr($template, [
       '%size%' => $sizes[array_rand($sizes)],
       '%type%' => $type,
       '%status%' => $statuses[array_rand($statuses)],
-    ]);
+    ]).' The coffee machine says: '.$message;
   }
 }
diff --git a/modules/custom/custom/coffee_shop/src/Service/CoffeeMachine.php b/modules/custom/custom/coffee_shop/src/Service/CoffeeMachine.php
new file mode 100644
index 00000000..da26f7ce
--- /dev/null
+++ b/modules/custom/custom/coffee_shop/src/Service/CoffeeMachine.php
@@ -0,0 +1,10 @@
+<?php
+
+namespace Drupal\coffee_shop\Service;
+
+class CoffeeMachine {
+  public function brew()
+  {
+    return 'coffee brewed!';
+  }
+}
