diff --git a/modules/custom/coffee_shop/src/Controller/CoffeeController.php b/modules/custom/coffee_shop/src/Controller/CoffeeController.php
index abd10aa8..7f29609e 100644
--- a/modules/custom/coffee_shop/src/Controller/CoffeeController.php
+++ b/modules/custom/coffee_shop/src/Controller/CoffeeController.php
@@ -7,11 +7,6 @@
 
 class CoffeeController extends ControllerBase {
   public function brewCoffee($type) {
-    if ($type === null) {
-      $type = $this->config('coffee_shop.default')
-        ->get('type');
-    }
-
     $barista =  \Drupal::getContainer()
       ->get('coffee_shop.barista');
     $text = $barista->prepareDrink($type);
diff --git a/modules/custom/coffee_shop/src/Service/Barista.php b/modules/custom/coffee_shop/src/Service/Barista.php
index 8ada2a9c..ae07b91c 100644
--- a/modules/custom/coffee_shop/src/Service/Barista.php
+++ b/modules/custom/coffee_shop/src/Service/Barista.php
@@ -3,7 +3,12 @@
 namespace Drupal\coffee_shop\Service;
 
 class Barista {
-  public function prepareDrink($type) {
+  public function prepareDrink($type = null) {
+    if ($type === null) {
+      $type = $this->config('coffee_shop.default')
+        ->get('type');
+    }
+
     $sizes = ['small', 'medium', 'large', 'scary huge'];
     $statuses = ['for here', 'to go', 'to sip while standing pensively'];
 
