diff --git a/modules/custom/coffee_shop/src/EventSubscriber/CoffeeDrinkerSubscriber.php b/modules/custom/coffee_shop/src/EventSubscriber/CoffeeDrinkerSubscriber.php
new file mode 100644
index 00000000..9b10b467
--- /dev/null
+++ b/modules/custom/coffee_shop/src/EventSubscriber/CoffeeDrinkerSubscriber.php
@@ -0,0 +1,7 @@
+<?php
+
+namespace Drupal\coffee_shop\EventSubscriber;
+
+class CoffeeDrinkerSubscriber {
+
+}
