diff --git a/modules/custom/coffee_shop/coffee_shop.services.yml b/modules/custom/coffee_shop/coffee_shop.services.yml
index 38d3bfdd..d7c3457c 100644
--- a/modules/custom/coffee_shop/coffee_shop.services.yml
+++ b/modules/custom/coffee_shop/coffee_shop.services.yml
@@ -8,3 +8,7 @@ services:
     arguments: ['@messenger']
     tags:
       - { name: event_subscriber }
+  coffee_shop.route_subscriber:
+    class: Drupal\coffee_shop\Routing\RouteSubscriber
+    tags:
+      - { name: event_subscriber }
diff --git a/modules/custom/coffee_shop/src/Routing/RouteSubscriber.php b/modules/custom/coffee_shop/src/Routing/RouteSubscriber.php
new file mode 100644
index 00000000..bc177089
--- /dev/null
+++ b/modules/custom/coffee_shop/src/Routing/RouteSubscriber.php
@@ -0,0 +1,20 @@
+<?php
+
+namespace Drupal\coffee_shop\Routing;
+
+use Drupal\Core\Routing\RouteSubscriberBase;
+use Symfony\Component\Routing\RouteCollection;
+
+/**
+ * Class RouteSubscriber.
+ *
+ * Listens to the dynamic route events.
+ */
+class RouteSubscriber extends RouteSubscriberBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function alterRoutes(RouteCollection $collection) {
+  }
+}
