diff --git a/modules/custom/coffee_shop/coffee_shop.services.yml b/modules/custom/coffee_shop/coffee_shop.services.yml
index 19adc132..1f000a7f 100644
--- a/modules/custom/coffee_shop/coffee_shop.services.yml
+++ b/modules/custom/coffee_shop/coffee_shop.services.yml
@@ -2,3 +2,7 @@ services:
   coffee_shop.barista:
     class: Drupal\coffee_shop\Service\Barista
     arguments: ['@config.factory']
+
+  coffee_shop.coffee_drinker_subscriber:
+    class: Drupal\coffee_shop\EventSubscriber\CoffeeDrinkerSubscriber
+    arguments: []
diff --git a/modules/custom/coffee_shop/src/EventSubscriber/CoffeeDrinkerSubscriber.php b/modules/custom/coffee_shop/src/EventSubscriber/CoffeeDrinkerSubscriber.php
index 9b10b467..551dc5ee 100644
--- a/modules/custom/coffee_shop/src/EventSubscriber/CoffeeDrinkerSubscriber.php
+++ b/modules/custom/coffee_shop/src/EventSubscriber/CoffeeDrinkerSubscriber.php
@@ -2,6 +2,17 @@
 
 namespace Drupal\coffee_shop\EventSubscriber;
 
-class CoffeeDrinkerSubscriber {
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
 
+class CoffeeDrinkerSubscriber implements EventSubscriberInterface {
+  public function onKernelRequest()
+  {
+    drupal_set_message('Good morning request! Time for some coffee!');
+  }
+
+  public static function getSubscribedEvents() {
+    return [
+      'kernel.request' => 'onKernelRequest',
+    ];
+  }
 }
