diff --git a/modules/custom/coffee_shop/src/EventSubscriber/CoffeeDrinkerSubscriber.php b/modules/custom/coffee_shop/src/EventSubscriber/CoffeeDrinkerSubscriber.php
index bbb25fd9..793d792d 100644
--- a/modules/custom/coffee_shop/src/EventSubscriber/CoffeeDrinkerSubscriber.php
+++ b/modules/custom/coffee_shop/src/EventSubscriber/CoffeeDrinkerSubscriber.php
@@ -4,9 +4,11 @@
 
 use Drupal\Core\Messenger\MessengerInterface;
 use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
 
 class CoffeeDrinkerSubscriber implements EventSubscriberInterface {
   private $messenger;
+  private $startTime;
 
   public function __construct(MessengerInterface $messenger) {
     $this->messenger = $messenger;
@@ -14,12 +16,21 @@ public function __construct(MessengerInterface $messenger) {
 
   public function onKernelRequest()
   {
+    $this->startTime = microtime(true);
     $this->messenger->addStatus('Good morning request! Time for some coffee!');
   }
 
+  public function onKernelResponse(FilterResponseEvent $event) {
+    $duration = microtime(true) - $this->startTime;
+    $event->getResponse()
+        ->headers
+        ->set('X-Brew-Time', $duration * 1000);
+  }
+
   public static function getSubscribedEvents() {
     return [
       'kernel.request' => 'onKernelRequest',
+      'kernel.response' => 'onKernelResponse',
     ];
   }
 }
