diff --git a/modules/custom/coffee_shop/src/Plugin/Block/CoffeeBlock.php b/modules/custom/coffee_shop/src/Plugin/Block/CoffeeBlock.php
index 77c16fb3..ea6dab83 100644
--- a/modules/custom/coffee_shop/src/Plugin/Block/CoffeeBlock.php
+++ b/modules/custom/coffee_shop/src/Plugin/Block/CoffeeBlock.php
@@ -18,8 +18,11 @@ class CoffeeBlock extends BlockBase {
    * {@inheritdoc}
    */
   public function build() {
+    $barista =  \Drupal::getContainer()
+      ->get('coffee_shop.barista');
+
     $build = [];
-    $build['coffee_block']['#markup'] = 'Implement CoffeeBlock.';
+    $build['coffee_block']['#markup'] = $barista->prepareDrink();
 
     return $build;
   }
