diff --git a/modules/custom/custom/coffee_shop/coffee_shop.services.yml b/modules/custom/custom/coffee_shop/coffee_shop.services.yml
index 11f22b6d..ed05c68e 100644
--- a/modules/custom/custom/coffee_shop/coffee_shop.services.yml
+++ b/modules/custom/custom/coffee_shop/coffee_shop.services.yml
@@ -5,8 +5,4 @@ services:
 #  _defaults:
 #    autowire: true
 
-  Drupal\coffee_shop\Service\Barista:
-    arguments: []
-
-  Drupal\coffee_shop\Service\CoffeeMachine:
-    arguments: []
+# just crickets here!
\ No newline at end of file
diff --git a/modules/custom/custom/coffee_shop/src/CoffeeShopServiceProvider.php b/modules/custom/custom/coffee_shop/src/CoffeeShopServiceProvider.php
index 67e18214..5012e310 100644
--- a/modules/custom/custom/coffee_shop/src/CoffeeShopServiceProvider.php
+++ b/modules/custom/custom/coffee_shop/src/CoffeeShopServiceProvider.php
@@ -4,15 +4,28 @@
 
 use Drupal\Core\DependencyInjection\ContainerBuilder;
 use Drupal\Core\DependencyInjection\ServiceProviderInterface;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\Finder\Finder;
 
 class CoffeeShopServiceProvider implements ServiceProviderInterface {
   public function register(ContainerBuilder $container) {
-    foreach ($container->getDefinitions() as $id => $definition) {
-      if (strpos($id, 'Drupal\\coffee_shop') !== 0) {
+    $finder = new Finder();
+    $finder->in(__DIR__.'/Service')
+      ->files()
+      ->name('*.php');
+
+    foreach ($finder as $fileInfo) {
+      $class = 'Drupal\coffee_shop\Service\\'
+        .substr($fileInfo->getFilename(), 0, -4);
+
+      // don't override any existing service
+      if ($container->hasDefinition($class)) {
         continue;
       }
 
+      $definition = new Definition($class);
       $definition->setAutowired(true);
+      $container->setDefinition($class, $definition);
     }
   }
 }
\ No newline at end of file
