diff --git a/modules/custom/coffee_shop/coffee_shop.routing.yml b/modules/custom/coffee_shop/coffee_shop.routing.yml
index b7cb073b..bc2f2ff1 100644
--- a/modules/custom/coffee_shop/coffee_shop.routing.yml
+++ b/modules/custom/coffee_shop/coffee_shop.routing.yml
@@ -1,5 +1,5 @@
 coffee_shop.coffee.brew:
-  path: /coffee/brew
+  path: /coffee/brew/{type}
   defaults:
     _controller: Drupal\coffee_shop\Controller\CoffeeController::brewCoffee
   requirements:
diff --git a/modules/custom/coffee_shop/src/Controller/CoffeeController.php b/modules/custom/coffee_shop/src/Controller/CoffeeController.php
index f3f13d41..c06a608b 100644
--- a/modules/custom/coffee_shop/src/Controller/CoffeeController.php
+++ b/modules/custom/coffee_shop/src/Controller/CoffeeController.php
@@ -5,7 +5,7 @@
 use Symfony\Component\HttpFoundation\Response;
 
 class CoffeeController {
-  public function brewCoffee() {
-    return new Response('Ding! Your delicious coffee is ready!');
+  public function brewCoffee($type) {
+    return new Response(sprintf('Ding! Your delicious %s is ready!', $type));
   }
 }
