diff --git a/modules/custom/coffee_shop/coffee_shop.links.menu.yml b/modules/custom/coffee_shop/coffee_shop.links.menu.yml
new file mode 100644
index 00000000..c5b3e77f
--- /dev/null
+++ b/modules/custom/coffee_shop/coffee_shop.links.menu.yml
@@ -0,0 +1,7 @@
+coffee_shop.default_form:
+  title: 'Coffee Form'
+  route_name: coffee_shop.default_form
+  description: 'Help brew delicious coffee'
+  parent: system.admin_config_system
+  weight: 99
+
diff --git a/modules/custom/coffee_shop/coffee_shop.routing.yml b/modules/custom/coffee_shop/coffee_shop.routing.yml
index f76b8eab..6ee0a51f 100644
--- a/modules/custom/coffee_shop/coffee_shop.routing.yml
+++ b/modules/custom/coffee_shop/coffee_shop.routing.yml
@@ -6,3 +6,14 @@ coffee_shop.coffee.brew:
     _title: 'Brew some coffee!'
   requirements:
     _permission: 'access content'
+
+coffee_shop.default_form:
+  path: '/admin/config/coffee_shop/default'
+  defaults:
+    _form: '\Drupal\coffee_shop\Form\DefaultForm'
+    _title: 'DefaultForm'
+  requirements:
+    _permission: 'access administration pages'
+  options:
+    _admin_route: TRUE
+
diff --git a/modules/custom/coffee_shop/config/install/coffee_shop.default.yml b/modules/custom/coffee_shop/config/install/coffee_shop.default.yml
new file mode 100644
index 00000000..6d8c314e
--- /dev/null
+++ b/modules/custom/coffee_shop/config/install/coffee_shop.default.yml
@@ -0,0 +1 @@
+coffee_shop:
diff --git a/modules/custom/coffee_shop/src/Form/DefaultForm.php b/modules/custom/coffee_shop/src/Form/DefaultForm.php
new file mode 100644
index 00000000..df8bc5a9
--- /dev/null
+++ b/modules/custom/coffee_shop/src/Form/DefaultForm.php
@@ -0,0 +1,62 @@
+<?php
+
+namespace Drupal\coffee_shop\Form;
+
+use Drupal\Core\Form\ConfigFormBase;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Class DefaultForm.
+ */
+class DefaultForm extends ConfigFormBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getEditableConfigNames() {
+    return [
+      'coffee_shop.default',
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'default_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $config = $this->config('coffee_shop.default');
+    $form['type'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Type'),
+      '#maxlength' => 64,
+      '#size' => 64,
+      '#default_value' => $config->get('type'),
+    ];
+    return parent::buildForm($form, $form_state);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validateForm(array &$form, FormStateInterface $form_state) {
+    parent::validateForm($form, $form_state);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    parent::submitForm($form, $form_state);
+
+    $this->config('coffee_shop.default')
+      ->set('type', $form_state->getValue('type'))
+      ->save();
+  }
+
+}
