diff --git a/modules/custom/custom/coffee_shop/src/Controller/CoffeeController.php b/modules/custom/custom/coffee_shop/src/Controller/CoffeeController.php
index d436b0ff..ed4fadae 100644
--- a/modules/custom/custom/coffee_shop/src/Controller/CoffeeController.php
+++ b/modules/custom/custom/coffee_shop/src/Controller/CoffeeController.php
@@ -2,10 +2,12 @@
 
 namespace Drupal\coffee_shop\Controller;
 
+use Drupal\coffee_shop\Service\Barista;
+
 class CoffeeController {
   public function brewCoffee($type) {
     $barista =  \Drupal::getContainer()
-      ->get('Drupal\coffee_shop\Service\Barista');
+      ->get(Barista::class);
     $text = $barista->prepareDrink($type);
 
     return [
