diff --git a/modules/custom/coffee_shop/coffee_shop.services.yml b/modules/custom/coffee_shop/coffee_shop.services.yml
new file mode 100644
index 00000000..e233c645
--- /dev/null
+++ b/modules/custom/coffee_shop/coffee_shop.services.yml
@@ -0,0 +1,4 @@
+services:
+  coffee_shop.barista:
+    class: Drupal\coffee_shop\Service\Barista
+    arguments: []
diff --git a/modules/custom/coffee_shop/src/Controller/CoffeeController.php b/modules/custom/coffee_shop/src/Controller/CoffeeController.php
index 3943e4ce..abd10aa8 100644
--- a/modules/custom/coffee_shop/src/Controller/CoffeeController.php
+++ b/modules/custom/coffee_shop/src/Controller/CoffeeController.php
@@ -12,7 +12,8 @@ public function brewCoffee($type) {
         ->get('type');
     }
 
-    $barista = new Barista();
+    $barista =  \Drupal::getContainer()
+      ->get('coffee_shop.barista');
     $text = $barista->prepareDrink($type);
 
     return [
