diff --git a/modules/custom/coffee_shop/coffee_shop.routing.yml b/modules/custom/coffee_shop/coffee_shop.routing.yml
index 6ee0a51f..45329f73 100644
--- a/modules/custom/coffee_shop/coffee_shop.routing.yml
+++ b/modules/custom/coffee_shop/coffee_shop.routing.yml
@@ -2,7 +2,7 @@ coffee_shop.coffee.brew:
   path: /coffee/brew/{type}
   defaults:
     _controller: Drupal\coffee_shop\Controller\CoffeeController::brewCoffee
-    type: coffee
+    type: null
     _title: 'Brew some coffee!'
   requirements:
     _permission: 'access content'
diff --git a/modules/custom/coffee_shop/src/Controller/CoffeeController.php b/modules/custom/coffee_shop/src/Controller/CoffeeController.php
index 21b2973d..d90d1f64 100644
--- a/modules/custom/coffee_shop/src/Controller/CoffeeController.php
+++ b/modules/custom/coffee_shop/src/Controller/CoffeeController.php
@@ -2,8 +2,15 @@
 
 namespace Drupal\coffee_shop\Controller;
 
-class CoffeeController {
+use Drupal\Core\Controller\ControllerBase;
+
+class CoffeeController extends ControllerBase {
   public function brewCoffee($type) {
+    if ($type === null) {
+      $type = $this->config('coffee_shop.default')
+        ->get('type');
+    }
+
     $text = sprintf('Ding! Your delicious %s is ready!', $type);
 
     return [
